listeners/InteractionCreateListener.go

```diff
@@ -160,6 +166,208 @@ func (h InteractionCreateListener) handleMessageComponents(ctx context.Context,
 		}
 	case "accept", "reject":
 		h.handleAcceptDeclineButtons(ctx, s, i)
+	case "unconditionalgiveawayjoin":
```

Comment:
giveaway_join and then just add if giveway.Levels != nil {} and handle your logic based on that